src/main/scala/org/ergoplatform/settings/ErgoValidationSettings.scala

```diff
@@ -84,20 +94,86 @@ object ErgoValidationSettings {
     * To be used during genesis state creation or to perform checks that are not allowed
     * to be deactivated via soft-forks.
     */
-  val initial: ErgoValidationSettings = new ErgoValidationSettings(ValidationRules.rulesSpec)
+  val initial: ErgoValidationSettings = new ErgoValidationSettings(ValidationRules.rulesSpec, Map.empty)
+
+  def parseBytesByPrefix(extension: ExtensionCandidate, prefix: Byte): Array[Byte] = {
+    val values = extension.fields
+        .filter(_._1(0) == prefix)
+        .sortBy(_._1(1))
+        .map(_._2)
+    Helpers.concatArrays(values)
+  }
 
   /**
     * Extracts ErgoValidationSettings from extension section of the block
     */
   def parseExtension(extension: ExtensionCandidate): Try[ErgoValidationSettings] = Try {
-    val bytes = extension.fields.filter(_._1.head == Extension.ValidationRulesPrefix).sortBy(_._1.last).flatMap(_._2)
-    ErgoValidationSettingsSerializer.parseBytes(bytes.toArray)
+    val ergoRulesBytes = parseBytesByPrefix(extension, Extension.ValidationRulesPrefix)
+    val ergoVSettings = ErgoValidationSettingsSerializer.parseBytes(ergoRulesBytes)
+    val sigmaRulesBytes = parseBytesByPrefix(extension, Extension.SigmaValidationRulesPrefix)
+    if (sigmaRulesBytes.nonEmpty) {
+      val r = SigmaSerializer.startReader(sigmaRulesBytes)
+      val nRules = r.getUShort()
+      val sigmaRules = (0 until nRules).map { _ =>
+        val ruleId = (r.getUShort() + RuleStatusSerializer.FirstRuleId).toShort
+        val status = RuleStatusSerializer.parse(r)
+        ruleId -> status
+      }
+      ergoVSettings.copy(sigmaRules = sigmaRules.toMap)
+    } else {
+      ergoVSettings
+    }
   }
+
+  object RuleStatusSerializer extends SigmaSerializer[SRuleStatus, SRuleStatus] {
+    val DisabledRuleCode = 1
+    val ReplacedRuleCode = 2
+    val ChangedRuleCode = 3
+    val FirstRuleId = 1000
+
+    override def serialize(obj: SRuleStatus, w: SigmaByteWriter): Unit = {
```

Comment:
Looks like this serializer does not satisfy our general serialization roundtrip check